src/expr.rs

```diff
@@ -35,7 +37,222 @@ pub enum Expr {
     Geometry(Geometry),
 }
 
+impl From<Value> for Expr {
+    fn from(v: Value) -> Expr {
+        let e: Expr = serde_json::from_value(v).unwrap();
+        e
+    }
+}
+
+impl From<Expr> for Value {
+    fn from(v: Expr) -> Value {
+        let v: Value = serde_json::to_value(v).unwrap();
+        v
+    }
+}
+
+
+impl TryInto<f64> for Expr {
+    type Error = ();
+    fn try_into(self) -> Result<f64, Self::Error> {
+        match self {
+            Expr::Float(v) => Ok(v),
+            Expr::Literal(v) => f64::from_str(&v).or(Err(())),
+            _ => Err(()),
+        }
+    }
+}
+
+impl TryInto<bool> for Expr {
+    type Error = ();
+    fn try_into(self) -> Result<bool, Self::Error> {
+        match self {
+            Expr::Bool(v) => Ok(v),
+            _ => Err(()),
+        }
+    }
+}
+
 impl Expr {
+    /// Insert values from properties from json
+    ///
+    ///  # Examples
+    ///
+    /// ```
+    /// use serde_json::{json, Value};
+    /// use cql2::Expr;
+    /// let item = json!({"properties":{"eo:cloud_cover":10, "datetime": "2020-01-01 00:00:00Z", "boolfield": true}});
+    /// let mut expr_json = json!(
+    ///     {
+    ///         "op": "+",
+    ///         "args": [
+    ///             {"property": "eo:cloud_cover"},
+    ///             10
+    ///         ]
+    ///     }
+    /// );
+    ///
+    /// let mut expr: Expr = serde_json::from_value(expr_json).unwrap();
+    /// println!("Initial {:?}", expr);
+    /// expr.reduce(&item);
+    ///
+    /// let output: f64;
+    /// if let Expr::Float(v) = expr {
+    ///     output = v;
+    /// } else {
+    ///     assert!(false);
+    ///     output = 0.0;
+    /// }
+    /// println!("Modified {:?}", expr);
+    ///
+    /// assert_eq!(20.0, output);
+    ///
+    ///
+    /// ```
+    pub fn reduce(&mut self, j: &Value) {
+        match self {
+            Expr::Property { property } => {
+                let mut prefixproperty: String = "properties.".to_string();
+                prefixproperty.push_str(property);
+
+                let mut propexpr: Option<Value> = None;
+                if j.dot_has(property) {
+                    propexpr = j.dot_get(property).unwrap();
+                } else {
+                    let mut prefixproperty: String = "properties.".to_string();
+                    prefixproperty.push_str(property);
+                    propexpr = j.dot_get(&prefixproperty).unwrap();
+                }
```

Comment:
Use some `Result` methods to simplify this, use `format!` instead of pushing onto the string, and don't unwrap:

```suggestion
                let propexpr = j.dot_get(property).or_else(|_| j.dot_get(&format!("properties.{}", property)))?;
```